pyspecProcScripts/first_level/QESR_rescale.py

```diff
@@ -1,24 +1,34 @@
 from pyspecdata.datadir import pyspec_config # piggyback on _pyspecdata
 from pint import UnitRegistry
 from numpy import sqrt
-default_Q = float(pyspec_config.get_setting("default Q", default=4700))
-# note that this will fail if you don't have "QESR Conversion" set in your pyspecdata file
-dint_conversion = pyspec_config.get_setting("QESR Conversion")
-if dint_conversion is not None:
-    dint_conversion = float(dint_conversion)
+class calib_info (object):
+    "class for caching calibration info"
+    def __init__():
+        self.current_calib_name = calibration_name
+    def use_calibration(calibration_name):
+        if calibration_name != self.current_calib_name:
+            assert type(calibration_name) is str, "the calibration name must be specified as a string!"
+            assert len(calibration_name)>0, "you MUST use a calibration name and set the values `[calibration_name] q` and `[calibration_name] conversion` in your _pyspecdata file!"
+            self.default_Q = float(pyspec_config.get_setting(f"{calibration_name} Q"))
+            # note that this will fail if you don't have "QESR Conversion" set in your pyspecdata file
+            self.dint_conversion = float(pyspec_config.get_setting(f"{calibration_name} Conversion"))
+            if dint_conversion is not None:
+                dint_conversion = float(dint_conversion)
+calibcache = calib_info()
```

Comment:
the point of this class construct is just so that we don't have to go back and re-read the settings file every time we ask for the calibration file (that's what I mean by "caching")